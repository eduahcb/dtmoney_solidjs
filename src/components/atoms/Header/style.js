import { styled } from 'solid-styled-components'

export const Container = styled.header`
  background-color: var(--blue);
`

export const Content = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  max-width: 1120px;
  margin: 0 auto;
  
  padding: 32px 16px 192px;
`
