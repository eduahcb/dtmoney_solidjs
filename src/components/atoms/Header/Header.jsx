
import { Container, Content } from './style'

export const Header = (props) => {
  return (
    <Container>
      <Content>
        {props.children}
      </Content>
    </Container>
  )
}
