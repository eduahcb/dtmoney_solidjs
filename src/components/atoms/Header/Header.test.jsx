import { describe, test, expect } from 'vitest'

import { render, screen } from '@solidjs/testing-library'

import { Header } from './Header'

describe('Header', () => {
  test('renders', () => {
    render(() => <Header><p>this is a text</p></Header>)

    expect(screen.getByText('this is a text')).toBeInTheDocument()
  })

  test('match snapshot', () => {
    const { asFragment } = render(() => <Header><p>this is a text</p></Header>)

    expect(asFragment()).toMatchSnapshot()
  })
})
