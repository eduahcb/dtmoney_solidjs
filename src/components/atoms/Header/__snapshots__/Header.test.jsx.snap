// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > match snapshot 1`] = `"<header class=\\"go1825331116\\"><div class=\\"go4273203935\\"><p>this is a text</p></div></header>"`;
